proxy/http/HttpSM.cc

```diff
@@ -7737,14 +7737,14 @@ HttpSM::redirect_request(const char *redirect_url, const int redirect_len)
 
     if (tmpOrigHost) {
       memcpy(origHost, tmpOrigHost, origHost_len);
-      origHost[origHost_len] = '\0';
+      origHost[std::min(origHost_len, MAXDNAME - 1)] = '\0';
     } else {
       valid_origHost = false;
     }
 
     char *tmpOrigMethod = (char *)t_state.hdr_info.server_request.method_get(&origMethod_len);
     if (tmpOrigMethod) {
-      memcpy(origMethod, tmpOrigMethod, origMethod_len);
+      memcpy(origMethod, tmpOrigMethod, std::min(origMethod_len, 255));
```

Comment:
Hmm, 255 seems kinda arbitrary... shouldn’t there be a define for this, or a sizeof?

Reply:
Yes. Will fix.